core/src/main/java/org/eclipse/hono/util/DataVolumePeriod.java

```diff
@@ -0,0 +1,75 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.util;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+
+/**
+ * Data volume period definition of the tenant resource limits .
+ */
+@JsonInclude(Include.NON_DEFAULT)
+public class DataVolumePeriod {
+
+    @JsonProperty(value = TenantConstants.FIELD_PERIOD_MODE, required = true)
+    private String mode;
```

Comment:
we should probably use an enum for the supported modes, don't you think?

Reply:
Its a good point and I thought about it too. The supported modes by the reference limit checks implementation are _days_ and _monthly_. If any user decides to support more modes like _yearly_ or _quarterly_ it would be easier if this is just a string and he/she can easily handle in their custom limit checks implementation. Thats why I didn't change it to enum.

Reply:
I see, makes sense to me.